README.md

```diff
@@ -1 +1,238 @@
+
+```
+ $ node main-environment
+URL is http://webapplog.com
+API key is D1FCFE58-575F-40FF-AF36-0CBD6A367756
+$ NODE_ENV=production node main-environment
+URL is http://webapplog.com
+API key is 34EC5CE9-CD02-4C4E-B5D9-D7CEF2F12F2C
+$ NODE_ENV=testing node main-environment
+URL is http://webapplog.com
+API key is D2230FC4-7E2A-4003-9372-0BF80107558D
+```
+
+To summarize the difference between objects and functions, the latter are more dynamic and allow for the so called initializer or constructor code, i.e., some code to tweak an object before exporting it.
+ 
+## module.exports and exports
+
+Simply put, `module.exports` is an alias to the `exports`. You can save your self a repetitive stress injury by using `exports`. Take a look at this module which uses `module.exports`:
+
+```js
+module.exports.sayHelloInEnglish = function() {
+  return 'Hello'
+}
+
+module.exports.sayHelloInTatar = function() {
+  return 'Isänmesez'
+}
+
+module.exports.sayHelloInSpanish = function() {
+  return 'Hola'
+}
+
+module.exports.sayHelloInJapanese = function() {
+  return 'こんにちは' // Kon'nichiwa
+}
+```
+
+The following version is an absolute exact equivalent:
+
+```js
+exports.sayHelloInEnglish = function() {
+  return 'Hello'
+}
+
+exports.sayHelloInTatar = function() {
+  return 'Isänmesez'
+}
+
+exports.sayHelloInSpanish = function() {
+  return 'Hola'
+}
+
+exports.sayHelloInJapanese = function() {
+  return 'こんにちは' // Kon'nichiwa
+}
+```
+
+That's great! What if we want to assign the whole object like we did with the configuration module?     Obviously the code with `module.exports` works because we used it in the previous section:
+
+```js
+module.exports = function(env) {
+  env = env || process.env.NODE_ENV
+  if (env == 'production') {
+    ...
+  } else if (env == 'testing') {
+    ...
+  } else { // Assume development
+    ...
+  }
+}
+```
+
+What about this code with `exports`? Will it work too? 
+
+```js
+exports = function(env) {
+  env = env || process.env.NODE_ENV
+  if (env == 'production') {
+    ...
+  } else if (env == 'testing') {
+    ...
+  } else { // Assume development
+    ...
+  }
+}
+```
+
+Go ahead and try to run it (`environment-exports.js`) with this one-liner command:
+
+```
+$ node -e "console.log(require('./environment-exports')().apiKey)"
+```
+
+If you're getting `TypeError: require(...) is not a function` then that's because we cannot re-assign exports just by itself. We need to use `module.exports =...` if we want to reassign the entire module. 
+
+When to use `module.exports = ...` vs. `exports.name = ...` or `module.exports.name = ...`? Node developers tend to use the former when you have a large class which take the whole file and the latter when you have a set of individual methods or objects. A good example will be an Express.js library which uses the former style and a collection of some utilities which are categorically the same but have distinct functionalities. 
```

Comment:
what do you mean by reassign? I didn't think we were reassigning anything if we're just requiring a different module?